File: two_sum/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapped_hashmap: dict = {}

        for i, num in enumerate(nums):
            complement: int = target - num 
            if complement in mapped_hashmap:
                return [mapped_hashmap[complement], i]
            mapped_hashmap[num] = i

File: two_sum/twoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mappedInputs = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int complement = target - nums[i];
            if (mappedInputs.containsKey(complement)) {
                return new int[]{mappedInputs.get(complement), i};
            }
            mappedInputs.put(nums[i], i);
        }
        return new int[]{};
    }
}